src/components/fields/date-input/date-input.tsx

```diff
@@ -30,13 +30,34 @@ export const DateInput = (props: IGenericFieldProps<IDateInputSchema>) => {
 	// EFFECTS
 	// =============================================================================
 	useEffect(() => {
+		const futureRule = validation?.find((rule) => "future" in rule);
+		const pastRule = validation?.find((rule) => "past" in rule);
+		const notFutureRule = validation?.find((rule) => "notFuture" in rule);
+		const notPastRule = validation?.find((rule) => "notPast" in rule);
 		setFieldValidationConfig(
 			id,
-			Yup.string().test("is-date", "Invalid date", (value) => {
-				if (!value || value === "") return true;
-				const date = new Date(value);
-				return !isNaN(date.valueOf());
-			}),
+			Yup.string()
+				.test("is-date", "Invalid date", (value) => {
+					if (!value || value === "") return true;
+					const date = new Date(value);
+					return !isNaN(date.valueOf());
+				})
+				.test("future", futureRule?.errorMessage || "Date must be in the future.", (value) => {
```

Comment:
note: these are rules applicable to date input field only, that's why they are added here